GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs

```diff
@@ -763,6 +773,398 @@ private List<int> GetStepPosition(DataRow[] allSteps, DataRow stepData, List<int
+
+            if (endpoint.Contains("{"))
+            {
+                stepData.Options = options;
+                return;
+            }
+            
+            questionKey = stringReplacementsService.DoReplacements(questionKey, apiExtraData);
+            questionKey = await ReplaceConfiguratorItemsAsync(questionKey, configuration, false);
+            questionKey = await stringReplacementsService.DoAllReplacementsAsync(questionKey, removeUnknownVariables: false);
+            
+            answersKey = stringReplacementsService.DoReplacements(answersKey, apiExtraData);
+            answersKey = await ReplaceConfiguratorItemsAsync(answersKey, configuration, false);
+            answersKey = await stringReplacementsService.DoAllReplacementsAsync(answersKey, removeUnknownVariables: false);
+            
+            var requestMethod = (Method) questionApi.GetDetailValue<int>("request_type");
+            
+            var restClient = new RestClient();
+            var restRequest = new RestRequest(endpoint, requestMethod);
+            
+            // If a body is provided perform replacements and add it to the request.
+            if (!String.IsNullOrWhiteSpace(requestJson))
+            {
+                requestJson = await ReplaceConfiguratorItemsAsync(requestJson, configuration, false);
+                requestJson = await ReplaceConfiguratorItemsAsync(requestJson, configuration, false);
+                requestJson = await stringReplacementsService.DoAllReplacementsAsync(requestJson, extraData, removeUnknownVariables: false);
+
+                var regex = new Regex("([\"'])?{[^\\]}\\s]*}([\"'])?", RegexOptions.Compiled | RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(2000));
+                requestJson = regex.Replace(requestJson, "null");
+
+                restRequest.AddBody(requestJson, MediaTypeNames.Application.Json);
+            }
+            
+            await AddAuthenticationToApiCall(restRequest, questionApi);
+            await AddAcceptLanguageToApiCall(restRequest);
+            await AddCustomHeadersAsync(restRequest, questionApi, vueConfiguration: configuration, extraData: extraData);
+            
+            var restResponse = await DoExternalConfiguratorApiCallAsync(restClient, restRequest);
+            if (!restResponse.IsSuccessful || restResponse.Content == null)
+            {
+                logger.LogWarning("Error while trying to get the configuration from API ({questionApi}). The API response HTTP code was '{restResponseStatusCode}' and the result was: {restResponseContent}.\n\n{requestJson}", questionApi.Title, restResponse.StatusCode, restResponse.Content, requestJson);
+                stepData.Options = options;
+                return;
+            }
+            
+            // Get the question from the response.
+            JToken responseData = restResponse.Content.StartsWith("{") ? JObject.Parse(restResponse.Content) : JArray.Parse(restResponse.Content);
```

Comment:
You should just use `JToken.Parse` here, that will work for any kind of JSON, arrays, objects etc. See here for more information: https://www.newtonsoft.com/json/help/html/ParseJsonAny.htm